### ISSUE TITLE
test: docker-compose cannot set up Kafka Cluster successfully now 

### ISSUE THREAD

**Issue by napallday** [IBM/sarama#2564]
##### Versions

*Please specify real version numbers or git SHAs, not just "Latest" since that changes fairly regularly.*

| Sarama | Kafka | Go |
|--------|-------|----|
|  v1.40.1 |  |  |

##### Logs

When filing an issue please provide logs from Sarama and Kafka if at all
possible. You can set `sarama.Logger` to a `log.Logger` to capture Sarama debug
output.

````
2023-08-05 15:13:06 advertised.listeners=LISTENER_INTERNAL://kafka-1:9091,LISTENER_LOCAL://localhost:29091
2023-08-05 15:13:06 auto.create.topics.enable=false
2023-08-05 15:13:06 broker.id=1
2023-08-05 15:13:06 broker.rack=1
2023-08-05 15:13:06 default.replication.factor=2
2023-08-05 15:13:06 delete.topic.enable=true
2023-08-05 15:13:06 group.initial.rebalance.delay.ms=0
2023-08-05 15:13:06 inter.broker.listener.name=LISTENER_INTERNAL
2023-08-05 15:13:06 listener.security.protocol.map=LISTENER_INTERNAL:PLAINTEXT,LISTENER_LOCAL:PLAINTEXT
2023-08-05 15:13:06 listeners=LISTENER_INTERNAL://:9091,LISTENER_LOCAL://:29091
2023-08-05 15:13:06 log.dirs=/tmp/kafka-logs
2023-08-05 15:13:06 log.retention.check.interval.ms=300000
2023-08-05 15:13:06 log.retention.hours=168
2023-08-05 15:13:06 num.io.threads=8
2023-08-05 15:13:06 num.network.threads=3
2023-08-05 15:13:06 num.partitions=1
2023-08-05 15:13:06 num.recovery.threads.per.data.dir=1
2023-08-05 15:13:06 offsets.topic.replication.factor=1
2023-08-05 15:13:06 replica.selector.class=org.apache.kafka.common.replica.RackAwareReplicaSelector
2023-08-05 15:13:06 socket.receive.buffer.bytes=102400
2023-08-05 15:13:06 socket.request.max.bytes=104857600
2023-08-05 15:13:06 socket.send.buffer.bytes=102400
2023-08-05 15:13:06 transaction.state.log.min.isr=1
2023-08-05 15:13:06 transaction.state.log.replication.factor=2
2023-08-05 15:13:06 zookeeper.connect=zookeeper-1:2181,zookeeper-2:2181,zookeeper-3:2181
2023-08-05 15:13:06 zookeeper.connection.timeout.ms=6000
2023-08-05 15:13:06 zookeeper.session.timeout.ms=6000
2023-08-05 15:13:20 log4j:WARN Failed to set property [conversionPattern] to value "[%d] %p %m (%c)%n". 
2023-08-05 15:13:20 java.lang.reflect.InvocationTargetException
2023-08-05 15:13:20     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-08-05 15:13:20     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2023-08-05 15:13:20     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-08-05 15:13:20     at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2023-08-05 15:13:20     at org.apache.log4j.config.PropertySetter.setProperty(PropertySetter.java:228)
2023-08-05 15:13:20     at org.apache.log4j.config.PropertySetter.setProperty(PropertySetter.java:192)
2023-08-05 15:13:20     at org.apache.log4j.config.PropertySetter.setProperties(PropertySetter.java:157)
2023-08-05 15:13:20     at org.apache.log4j.config.PropertySetter.setProperties(PropertySetter.java:100)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.parseAppender(PropertyConfigurator.java:769)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.parseCategory(PropertyConfigurator.java:738)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.configureRootCategory(PropertyConfigurator.java:630)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:516)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:577)
2023-08-05 15:13:20     at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:504)
2023-08-05 15:13:20     at org.apache.log4j.LogManager.<clinit>(LogManager.java:119)
2023-08-05 15:13:20     at org.slf4j.impl.Reload4jLoggerFactory.<init>(Reload4jLoggerFactory.java:67)
2023-08-05 15:13:20     at org.slf4j.impl.StaticLoggerBinder.<init>(StaticLoggerBinder.java:72)
2023-08-05 15:13:20     at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:417)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:362)
2023-08-05 15:13:20     at com.typesafe.scalalogging.Logger$.apply(Logger.scala:31)
2023-08-05 15:13:20     at kafka.utils.Log4jControllerRegistration$.<init>(Logging.scala:25)
2023-08-05 15:13:20     at kafka.utils.Log4jControllerRegistration$.<clinit>(Logging.scala)
2023-08-05 15:13:20     at kafka.utils.Logging.$init$(Logging.scala:47)
2023-08-05 15:13:20     at kafka.Kafka$.<init>(Kafka.scala:30)
2023-08-05 15:13:20     at kafka.Kafka$.<clinit>(Kafka.scala)
2023-08-05 15:13:20     at kafka.Kafka.main(Kafka.scala)
2023-08-05 15:13:20 Caused by: java.lang.Error: java.io.FileNotFoundException: /usr/lib/jvm/java-11-openjdk-11.0.20.0.8-2.el8.x86_64/lib/tzdb.dat (No such file or directory)
2023-08-05 15:13:20     at java.base/sun.util.calendar.ZoneInfoFile$1.run(ZoneInfoFile.java:261)
2023-08-05 15:13:20     at java.base/sun.util.calendar.ZoneInfoFile$1.run(ZoneInfoFile.java:251)
2023-08-05 15:13:20     at java.base/java.security.AccessController.doPrivileged(Native Method)
2023-08-05 15:13:20     at java.base/sun.util.calendar.ZoneInfoFile.<clinit>(ZoneInfoFile.java:251)
2023-08-05 15:13:20     at java.base/sun.util.calendar.ZoneInfo.getTimeZone(ZoneInfo.java:588)
2023-08-05 15:13:20     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:577)
2023-08-05 15:13:20     at java.base/java.util.TimeZone.setDefaultZone(TimeZone.java:682)
2023-08-05 15:13:20     at java.base/java.util.TimeZone.getDefaultRef(TimeZone.java:653)
2023-08-05 15:13:20     at java.base/java.util.TimeZone.getDefault(TimeZone.java:642)
2023-08-05 15:13:20     at java.base/java.util.Calendar.defaultTimeZone(Calendar.java:1679)
2023-08-05 15:13:20     at java.base/java.util.Calendar.getInstance(Calendar.java:1629)
2023-08-05 15:13:20     at org.apache.log4j.helpers.AbsoluteTimeDateFormat.<init>(AbsoluteTimeDateFormat.java:61)
2023-08-05 15:13:20     at org.apache.log4j.helpers.ISO8601DateFormat.<init>(ISO8601DateFormat.java:45)
2023-08-05 15:13:20     at org.apache.log4j.helpers.PatternParser.finalizeConverter(PatternParser.java:247)
2023-08-05 15:13:20     at org.apache.log4j.helpers.PatternParser.parse(PatternParser.java:182)
2023-08-05 15:13:20     at org.apache.log4j.PatternLayout.setConversionPattern(PatternLayout.java:458)
2023-08-05 15:13:20     ... 29 more
2023-08-05 15:13:20 Caused by: java.io.FileNotFoundException: /usr/lib/jvm/java-11-openjdk-11.0.20.0.8-2.el8.x86_64/lib/tzdb.dat (No such file or directory)
2023-08-05 15:13:20     at java.base/java.io.FileInputStream.open0(Native Method)
2023-08-05 15:13:20     at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
2023-08-05 15:13:20     at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
2023-08-05 15:13:20     at java.base/sun.util.calendar.ZoneInfoFile$1.run(ZoneInfoFile.java:255)
2023-08-05 15:13:20     ... 44 more
2023-08-05 15:13:20 log4j:WARN Error during default initialization
2023-08-05 15:13:20 java.lang.NoClassDefFoundError: Could not initialize class sun.util.calendar.ZoneInfoFile
2023-08-05 15:13:20     at java.base/sun.util.calendar.ZoneInfo.getTimeZone(ZoneInfo.java:588)
2023-08-05 15:13:20     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:577)
2023-08-05 15:13:20     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:517)
2023-08-05 15:13:20     at org.apache.log4j.DailyRollingFileAppender.<clinit>(DailyRollingFileAppender.java:182)
2023-08-05 15:13:20     at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2023-08-05 15:13:20     at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2023-08-05 15:13:20     at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2023-08-05 15:13:20     at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
2023-08-05 15:13:20     at java.base/java.lang.Class.newInstance(Class.java:584)
2023-08-05 15:13:20     at org.apache.log4j.helpers.OptionConverter.instantiateByClassName(OptionConverter.java:313)
2023-08-05 15:13:20     at org.apache.log4j.helpers.OptionConverter.instantiateByKey(OptionConverter.java:123)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.parseAppender(PropertyConfigurator.java:755)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.parseCategory(PropertyConfigurator.java:738)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.configureRootCategory(PropertyConfigurator.java:630)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:516)
2023-08-05 15:13:20     at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:577)
2023-08-05 15:13:20     at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:504)
2023-08-05 15:13:20     at org.apache.log4j.LogManager.<clinit>(LogManager.java:119)
2023-08-05 15:13:20     at org.slf4j.impl.Reload4jLoggerFactory.<init>(Reload4jLoggerFactory.java:67)
2023-08-05 15:13:20     at org.slf4j.impl.StaticLoggerBinder.<init>(StaticLoggerBinder.java:72)
2023-08-05 15:13:20     at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:417)
2023-08-05 15:13:20     at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:362)
2023-08-05 15:13:20     at com.typesafe.scalalogging.Logger$.apply(Logger.scala:31)
2023-08-05 15:13:20     at kafka.utils.Log4jControllerRegistration$.<init>(Logging.scala:25)
2023-08-05 15:13:20     at kafka.utils.Log4jControllerRegistration$.<clinit>(Logging.scala)
2023-08-05 15:13:20     at kafka.utils.Logging.$init$(Logging.scala:47)
2023-08-05 15:13:20     at kafka.Kafka$.<init>(Kafka.scala:30)
2023-08-05 15:13:20     at kafka.Kafka$.<clinit>(Kafka.scala)
2023-08-05 15:13:20     at kafka.Kafka.main(Kafka.scala)
2023-08-05 15:13:20 Registered kafka:type=kafka.Log4jController MBean
2023-08-05 15:13:23 Setting -D jdk.tls.rejectClientInitiatedRenegotiation=true to disable client-initiated TLS renegotiation
2023-08-05 15:13:23 Registered signal handlers for TERM, INT, HUP
2023-08-05 15:13:23 starting
2023-08-05 15:13:23 Connecting to zookeeper on zookeeper-1:2181,zookeeper-2:2181,zookeeper-3:2181
2023-08-05 15:13:24 [ZooKeeperClient Kafka server] Initializing a new session to zookeeper-1:2181,zookeeper-2:2181,zookeeper-3:2181.
2023-08-05 15:13:24 Client environment:zookeeper.version=3.6.3--6401e4ad2087061bc6b9f80dec2d69f2e3c8660a, built on 04/08/2021 16:35 GMT
2023-08-05 15:13:24 Client environment:host.name=80174a374216
2023-08-05 15:13:24 Client environment:java.version=11.0.20
2023-08-05 15:13:24 Client environment:java.vendor=Red Hat, Inc.
2023-08-05 15:13:24 Client environment:java.home=/usr/lib/jvm/java-11-openjdk-11.0.20.0.8-2.el8.x86_64
2023-08-05 15:13:24 Client environment:java.class.path=/opt/kafka-3.3.2/bin/../libs/activation-1.1.1.jar:/opt/kafka-3.3.2/bin/../libs/aopalliance-repackaged-2.6.1.jar:/opt/kafka-3.3.2/bin/../libs/argparse4j-0.7.0.jar:/opt/kafka-3.3.2/bin/../libs/audience-annotations-0.5.0.jar:/opt/kafka-3.3.2/bin/../libs/commons-cli-1.4.jar:/opt/kafka-3.3.2/bin/../libs/commons-lang3-3.12.0.jar:/opt/kafka-3.3.2/bin/../libs/commons-lang3-3.8.1.jar:/opt/kafka-3.3.2/bin/../libs/connect-api-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/connect-basic-auth-extension-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/connect-json-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/connect-mirror-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/connect-mirror-client-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/connect-runtime-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/connect-transforms-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/hk2-api-2.6.1.jar:/opt/kafka-3.3.2/bin/../libs/hk2-locator-2.6.1.jar:/opt/kafka-3.3.2/bin/../libs/hk2-utils-2.6.1.jar:/opt/kafka-3.3.2/bin/../libs/jackson-annotations-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-core-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-databind-2.13.4.2.jar:/opt/kafka-3.3.2/bin/../libs/jackson-dataformat-csv-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-datatype-jdk8-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-jaxrs-base-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-jaxrs-json-provider-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-module-jaxb-annotations-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jackson-module-scala_2.12-2.13.4.jar:/opt/kafka-3.3.2/bin/../libs/jakarta.activation-api-1.2.2.jar:/opt/kafka-3.3.2/bin/../libs/jakarta.annotation-api-1.3.5.jar:/opt/kafka-3.3.2/bin/../libs/jakarta.inject-2.6.1.jar:/opt/kafka-3.3.2/bin/../libs/jakarta.validation-api-2.0.2.jar:/opt/kafka-3.3.2/bin/../libs/jakarta.ws.rs-api-2.1.6.jar:/opt/kafka-3.3.2/bin/../libs/jakarta.xml.bind-api-2.3.3.jar:/opt/kafka-3.3.2/bin/../libs/javassist-3.27.0-GA.jar:/opt/kafka-3.3.2/bin/../libs/javax.servlet-api-3.1.0.jar:/opt/kafka-3.3.2/bin/../libs/javax.ws.rs-api-2.1.1.jar:/opt/kafka-3.3.2/bin/../libs/jaxb-api-2.3.0.jar:/opt/kafka-3.3.2/bin/../libs/jersey-client-2.34.jar:/opt/kafka-3.3.2/bin/../libs/jersey-common-2.34.jar:/opt/kafka-3.3.2/bin/../libs/jersey-container-servlet-2.34.jar:/opt/kafka-3.3.2/bin/../libs/jersey-container-servlet-core-2.34.jar:/opt/kafka-3.3.2/bin/../libs/jersey-hk2-2.34.jar:/opt/kafka-3.3.2/bin/../libs/jersey-server-2.34.jar:/opt/kafka-3.3.2/bin/../libs/jetty-client-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-continuation-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-http-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-io-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-security-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-server-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-servlet-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-servlets-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-util-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jetty-util-ajax-9.4.48.v20220622.jar:/opt/kafka-3.3.2/bin/../libs/jline-3.21.0.jar:/opt/kafka-3.3.2/bin/../libs/jopt-simple-5.0.4.jar:/opt/kafka-3.3.2/bin/../libs/jose4j-0.7.9.jar:/opt/kafka-3.3.2/bin/../libs/kafka-clients-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-log4j-appender-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-metadata-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-raft-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-server-common-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-shell-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-storage-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-storage-api-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-streams-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-streams-examples-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-streams-scala_2.12-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-streams-test-utils-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka-tools-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/kafka_2.12-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/lz4-java-1.8.0.jar:/opt/kafka-3.3.2/bin/../libs/maven-artifact-3.8.4.jar:/opt/kafka-3.3.2/bin/../libs/metrics-core-2.2.0.jar:/opt/kafka-3.3.2/bin/../libs/metrics-core-4.1.12.1.jar:/opt/kafka-3.3.2/bin/../libs/netty-buffer-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-codec-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-common-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-handler-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-resolver-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-transport-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-transport-classes-epoll-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-transport-native-epoll-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/netty-transport-native-unix-common-4.1.78.Final.jar:/opt/kafka-3.3.2/bin/../libs/osgi-resource-locator-1.0.3.jar:/opt/kafka-3.3.2/bin/../libs/paranamer-2.8.jar:/opt/kafka-3.3.2/bin/../libs/plexus-utils-3.3.0.jar:/opt/kafka-3.3.2/bin/../libs/reflections-0.9.12.jar:/opt/kafka-3.3.2/bin/../libs/reload4j-1.2.19.jar:/opt/kafka-3.3.2/bin/../libs/rocksdbjni-7.1.2.jar:/opt/kafka-3.3.2/bin/../libs/scala-collection-compat_2.12-2.6.0.jar:/opt/kafka-3.3.2/bin/../libs/scala-java8-compat_2.12-1.0.2.jar:/opt/kafka-3.3.2/bin/../libs/scala-library-2.12.15.jar:/opt/kafka-3.3.2/bin/../libs/scala-logging_2.12-3.9.4.jar:/opt/kafka-3.3.2/bin/../libs/scala-reflect-2.12.15.jar:/opt/kafka-3.3.2/bin/../libs/slf4j-api-1.7.36.jar:/opt/kafka-3.3.2/bin/../libs/slf4j-reload4j-1.7.36.jar:/opt/kafka-3.3.2/bin/../libs/snappy-java-1.1.8.4.jar:/opt/kafka-3.3.2/bin/../libs/swagger-annotations-2.2.0.jar:/opt/kafka-3.3.2/bin/../libs/trogdor-3.3.2.jar:/opt/kafka-3.3.2/bin/../libs/zookeeper-3.6.3.jar:/opt/kafka-3.3.2/bin/../libs/zookeeper-jute-3.6.3.jar:/opt/kafka-3.3.2/bin/../libs/zstd-jni-1.5.2-1.jar
2023-08-05 15:13:24 Client environment:java.library.path=/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
2023-08-05 15:13:24 Client environment:java.io.tmpdir=/tmp
2023-08-05 15:13:24 Client environment:java.compiler=<NA>
2023-08-05 15:13:24 Client environment:os.name=Linux
2023-08-05 15:13:24 Client environment:os.arch=amd64
2023-08-05 15:13:24 Client environment:os.version=5.15.49-linuxkit
2023-08-05 15:13:24 Client environment:user.name=root
2023-08-05 15:13:24 Client environment:user.home=/root
2023-08-05 15:13:24 Client environment:user.dir=/opt/kafka-3.3.2
2023-08-05 15:13:24 Client environment:os.memory.free=973MB
2023-08-05 15:13:24 Client environment:os.memory.max=1024MB
2023-08-05 15:13:24 Client environment:os.memory.total=1024MB
2023-08-05 15:13:24 Initiating client connection, connectString=zookeeper-1:2181,zookeeper-2:2181,zookeeper-3:2181 sessionTimeout=6000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@495ee280
2023-08-05 15:13:24 jute.maxbuffer value is 4194304 Bytes
2023-08-05 15:13:24 zookeeper.request.timeout value is 0. feature enabled=false
2023-08-05 15:13:24 [ZooKeeperClient Kafka server] Waiting until connected.
2023-08-05 15:13:24 Opening socket connection to server zookeeper-1/192.168.240.8:2181.
2023-08-05 15:13:24 Socket connection established, initiating session, client: /192.168.240.6:35076, server: zookeeper-1/192.168.240.8:2181
2023-08-05 15:13:24 Session establishment complete on server zookeeper-1/192.168.240.8:2181, session id = 0x10009f45ec00002, negotiated timeout = 6000
2023-08-05 15:13:24 [ZooKeeperClient Kafka server] Connected.
2023-08-05 15:13:24 Fatal error during KafkaServer startup. Prepare to shutdown
2023-08-05 15:13:24 java.lang.NoClassDefFoundError: Could not initialize class sun.util.calendar.ZoneInfoFile
2023-08-05 15:13:24     at java.base/sun.util.calendar.ZoneInfo.getTimeZone(ZoneInfo.java:588)
2023-08-05 15:13:24     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:577)
2023-08-05 15:13:24     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:517)
2023-08-05 15:13:24     at com.fasterxml.jackson.databind.cfg.BaseSettings.<clinit>(BaseSettings.java:38)
2023-08-05 15:13:24     at com.fasterxml.jackson.databind.ObjectMapper.<clinit>(ObjectMapper.java:381)
2023-08-05 15:13:24     at kafka.utils.Json$.<init>(Json.scala:31)
2023-08-05 15:13:24     at kafka.utils.Json$.<clinit>(Json.scala)
2023-08-05 15:13:24     at kafka.zk.ClusterIdZNode$.toJson(ZkData.scala:775)
2023-08-05 15:13:24     at kafka.zk.KafkaZkClient.createOrGetClusterId(KafkaZkClient.scala:1588)
2023-08-05 15:13:24     at kafka.server.KafkaServer.$anonfun$getOrGenerateClusterId$1(KafkaServer.scala:508)
2023-08-05 15:13:24     at scala.Option.getOrElse(Option.scala:189)
2023-08-05 15:13:24     at kafka.server.KafkaServer.getOrGenerateClusterId(KafkaServer.scala:508)
2023-08-05 15:13:24     at kafka.server.KafkaServer.startup(KafkaServer.scala:207)
2023-08-05 15:13:24     at kafka.Kafka$.main(Kafka.scala:109)
2023-08-05 15:13:24     at kafka.Kafka.main(Kafka.scala)
2023-08-05 15:13:24 shutting down
2023-08-05 15:13:24 [ZooKeeperClient Kafka server] Closing.
2023-08-05 15:13:24 Session: 0x10009f45ec00002 closed
2023-08-05 15:13:24 EventThread shut down for session: 0x10009f45ec00002
2023-08-05 15:13:24 [ZooKeeperClient Kafka server] Closed.
2023-08-05 15:13:24 App info kafka.server for 1 unregistered
2023-08-05 15:13:24 shut down completed
2023-08-05 15:13:24 Exiting Kafka due to fatal exception during startup.
2023-08-05 15:13:24 java.lang.NoClassDefFoundError: Could not initialize class sun.util.calendar.ZoneInfoFile
2023-08-05 15:13:24     at java.base/sun.util.calendar.ZoneInfo.getTimeZone(ZoneInfo.java:588)
2023-08-05 15:13:24     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:577)
2023-08-05 15:13:24     at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:517)
2023-08-05 15:13:24     at com.fasterxml.jackson.databind.cfg.BaseSettings.<clinit>(BaseSettings.java:38)
2023-08-05 15:13:24     at com.fasterxml.jackson.databind.ObjectMapper.<clinit>(ObjectMapper.java:381)
2023-08-05 15:13:24     at kafka.utils.Json$.<init>(Json.scala:31)
2023-08-05 15:13:24     at kafka.utils.Json$.<clinit>(Json.scala)
2023-08-05 15:13:24     at kafka.zk.ClusterIdZNode$.toJson(ZkData.scala:775)
2023-08-05 15:13:24     at kafka.zk.KafkaZkClient.createOrGetClusterId(KafkaZkClient.scala:1588)
2023-08-05 15:13:24     at kafka.server.KafkaServer.$anonfun$getOrGenerateClusterId$1(KafkaServer.scala:508)
2023-08-05 15:13:24     at scala.Option.getOrElse(Option.scala:189)
2023-08-05 15:13:24     at kafka.server.KafkaServer.getOrGenerateClusterId(KafkaServer.scala:508)
2023-08-05 15:13:24     at kafka.server.KafkaServer.startup(KafkaServer.scala:207)
2023-08-05 15:13:24     at kafka.Kafka$.main(Kafka.scala:109)
2023-08-05 15:13:24     at kafka.Kafka.main(Kafka.scala)
2023-08-05 15:13:24 shutting down
````

**Comment by beanz** [comment]
This will need fix similar to f6ccc6f79ca917e41eefbcd9c1f5880105209d2a

**Comment by hindessm** [comment]
That fix is in 1.40.1 so you probably just need to do `docker-compose up --build` to force a fresh image to be created.

**Comment by napallday** [comment]
cool, it works! thanks for the prompt reply 😀 @hindessm
